MiniAn/minian_main.py

```diff
@@ -0,0 +1,355 @@
+            C_new, S_new, b0_new, c0_new, g, mask = update_temporal(A, C, YrA=YrA, **minian_parameters.params_update_temporal)
+
+        # Save
+        print(mn_defs.Messages.CNMF_SAVE_INTERMED.format("2nd"), flush=True)
+        C = save_minian(C_new.rename("C").chunk({"unit_id": 1, "frame": -1}), intpath, overwrite=True)
+        C_chk = save_minian(C.rename("C_chk"), intpath, overwrite=True, chunks={"unit_id": -1, "frame": chk["frame"]})
+        S = save_minian(S_new.rename("S").chunk({"unit_id": 1, "frame": -1}), intpath, overwrite=True)
+        b0 = save_minian(b0_new.rename("b0").chunk({"unit_id": 1, "frame": -1}), intpath, overwrite=True)
+        c0 = save_minian(c0_new.rename("c0").chunk({"unit_id": 1, "frame": -1}), intpath, overwrite=True)
+        A = A.sel(unit_id=C.coords["unit_id"].values)
+
+        AC = compute_AtC(A, C_chk)
+
+    ### Save final results ###
+    print(mn_defs.Messages.SAVING_FINAL, flush=True)
+    A = save_minian(A.rename("A"), **minian_parameters.params_save_minian)
+    C = save_minian(C.rename("C"), **minian_parameters.params_save_minian)
+    AC = save_minian(AC.rename("AC"), **minian_parameters.params_save_minian)
+    S = save_minian(S.rename("S"), **minian_parameters.params_save_minian)
+    c0 = save_minian(c0.rename("c0"), **minian_parameters.params_save_minian)
+    b0 = save_minian(b0.rename("b0"), **minian_parameters.params_save_minian)
+    b = save_minian(b.rename("b"), **minian_parameters.params_save_minian)
+    f = save_minian(f.rename("f"), **minian_parameters.params_save_minian)
+
+    ### Save results to doric file ###
+    print(mn_defs.Messages.SAVING_TO_DORIC, flush=True)
+    # Get the path from the source data
+    data, driver, operation, series, sensor = minian_parameters.get_hdf5path_struct()
+
+    # Get paramaters of the operation on source data
+    params_source_data = utils.load_attributes(file_, f"{data}/{driver}/{operation}")
+    # Get the attributes of the images stack
+    attrs = utils.load_attributes(file_, f"{minian_parameters.get_clean_h5path()}/ImagesStack")
+    file_.close()
+
+    # Parameters
+    # Set only "Operations" for params_srouce_data
+    if mn_defs.ParametersKeys.OPERATION_NAME in params_source_data:
+        if mn_defs.ParametersKeys.OPERATIONS not in params_source_data:
+            params_source_data[mn_defs.ParametersKeys.OPERATIONS] = params_source_data[mn_defs.ParametersKeys.OPERATION_NAME]
+
+        del params_source_data[mn_defs.ParametersKeys.OPERATION_NAME]
+
+    if minian_parameters.parameters[mn_defs.ParametersKeys.SPATIAL_DOWN_SAMP] > 1:
+        minian_parameters.parameters["BinningFactor"] = minian_parameters.parameters[mn_defs.ParametersKeys.SPATIAL_DOWN_SAMP]
+
+    mn_utils.save_minian_to_doric(
+        Y, A, C, AC, S,
+        fr = minian_parameters.fr,
+        bits_count = attrs['BitsCount'],
+        qt_format = attrs['Format'],
+        imagesStackUsername = attrs['Username'] if 'Username' in attrs else sensor,
```

Comment:
`username`

Reply:
@4D42 I think you missed some comments

Reply:
this comment too :/ sometime I don't see all the comments